rfcs/inline_symbols.md

```diff
@@ -0,0 +1,521 @@
+symbols, annotations, and field names that needed to be added to the symbol table and once to write
+the sensor data itself. However, because none of the values being added to the symbol table appear
+more than once, we do not benefit from the compactness of the symbol ID encodings. The overhead
+associated with adding each of them to the symbol table slows down the writing process and inflates
+the overall size of the data.
+
+| Format | Size in Bytes |
+|:--|:--:|
+| Text | 94 |
+| Binary | 100 |
+
+### Long-lived streams
+
+In long Ion streams, the symbol ID encoding mandate can cause the symbol table to become quite large.
+This causes two problems:
+
+1. Both Readers and Writers are required to hold the entire symbol table in memory. The larger it
+   becomes, the more memory this consumes.
+2. Symbol IDs are encoded as variable-length unsigned integers; higher ID numbers require more bytes
+   to encode.
+   
+The only way to remove entries from the active symbol table is to reset it or import a different one.
+In either case, we are likely to discard very valuable symbols along with those that are infrequently
+referenced. Re-adding the valuable symbols later wastes processing time and adds to the size of the
+stream. Being more selective about which symbols are added to the symbol table would mitigate these
+issues.
+
+### Serializing maps, dictionaries, and associative arrays
+
+Ion's `struct` data type closely resembles the classes, records, and structs found in a variety
+of programming languages. This makes it a natural choice as a serialized representation for such
+values.
+
+Consider this example of a Java class:
+
+```java
+public class Quux {
+  public String foo;
+  public String bar;
+  public int baz;
+
+  // ... rest of the class
+}
+```
+
+In Ion streams that contain several serialized instances of this class, we are guaranteed to recoup
+the data size cost of adding `foo`, `bar`, and `baz` to the symbol table. The class's fields are a
+fixed set, so every serialized instance will need to reference each field.
+
+Ion's `struct` data type also acts as a mapping from text keys to values of any data type. This
+closely (though often imperfectly) aligns with common data types available in a variety of
+programming languages, including Java's `Map<String, Object>`, JavaScript's `Object`, and so on.
+
+Consider this example of a Python dictionary:
+
+```python
+logins_by_user_id = {
+  'f354c586' : ["2020-11-05T12:01:17.900Z", "2020-11-06T21:08:16.965Z"],
+  'd3f712ea' : ["2019-07-21T11:24:55.000Z", "2020-05-11T19:44:01.044Z"],
+  '467cba51' : ["2020-10-03T01:51:30.011Z", "2020-10-29T22:40:40.331Z"],
+}
+```
+
+Unlike classes and their analogs, the keys used in a series of maps written to an Ion stream are not
+guaranteed to be the same over time. Maps using high-cardinality values like UUIDs and datetimes as
+keys will have keys that never repeat at all.
```

Comment:
Is `at all` needed here? Can this be left as `will have keys that never repeat.`?